apps/studio/components/interfaces/DiskManagement/DiskManagement.schema.ts

```diff
@@ -0,0 +1,169 @@
+          path: ['provisionedIOPS'],
+        })
+      }
+
+      const maxIOPSforDiskSizeWithio2 = calculateMaxIopsAllowedForDiskSizeWithio2(totalSize)
+
+      if (provisionedIOPS < IOPS_RANGE[DiskType.IO2].min) {
+        ctx.addIssue({
+          code: z.ZodIssueCode.custom,
+          message: `Provisioned IOPS must be at least ${formatNumber(IOPS_RANGE[DiskType.IO2].min)}`,
+          path: ['provisionedIOPS'],
+        })
+      } else if (provisionedIOPS > maxIOPSforDiskSizeWithio2) {
+        if (totalSize >= 8) {
+          const diskSizeRequiredForIopsWithIo2 =
+            calculateDiskSizeRequiredForIopsWithIo2(provisionedIOPS)
+
+          ctx.addIssue({
+            code: z.ZodIssueCode.custom,
+            message: `Larger Disk size of at least ${formatNumber(diskSizeRequiredForIopsWithIo2)} GB required. Current max is ${formatNumber(maxIOPSforDiskSizeWithio2)} IOPS.`,
+            path: ['provisionedIOPS'],
+          })
+        } else {
+          ctx.addIssue({
+            code: z.ZodIssueCode.custom,
+            message: `Invalid IOPS value due to small disk size`,
+            path: ['provisionedIOPS'],
+          })
+        }
+      }
+
+      if (throughput !== undefined) {
+        ctx.addIssue({
+          code: z.ZodIssueCode.custom,
+          message: 'Throughput is not configurable for io2.',
+          path: ['throughput'],
+        })
+      }
+
+      if (totalSize > DISK_LIMITS[DiskType.IO2].maxStorage) {
+        ctx.addIssue({
+          code: z.ZodIssueCode.custom,
+          message: `Allocated disksize must not exceed ${formatNumber(DISK_LIMITS[DiskType.IO2].maxStorage)} GB `,
+          path: ['totalSize'],
+        })
+      }
+    }
+
+    if (storageType === 'gp3') {
+      const maxIopsAllowedForDiskSizeWithGp3 = calculateMaxIopsAllowedForDiskSizeWithGp3(totalSize)
+
+      if (provisionedIOPS > IOPS_RANGE[DiskType.GP3].max) {
+        ctx.addIssue({
+          code: z.ZodIssueCode.custom,
+          message: `IOPS can not exceed ${formatNumber(IOPS_RANGE[DiskType.GP3].max)} for GP3 Disk. Change the Disk type to io2 for higher IOPS support.`,
+          path: ['provisionedIOPS'],
+        })
+      }
+
+      if (provisionedIOPS < IOPS_RANGE[DiskType.GP3].min) {
+        ctx.addIssue({
+          code: z.ZodIssueCode.custom,
+          message: `IOPS must be at least ${formatNumber(IOPS_RANGE[DiskType.GP3].min)}`,
+          path: ['provisionedIOPS'],
+        })
+      } else if (provisionedIOPS > maxIopsAllowedForDiskSizeWithGp3) {
+        if (totalSize >= 8) {
+          const diskSizeRequiredForIopsWithGp3 =
+            calculateDiskSizeRequiredForIopsWithGp3(provisionedIOPS)
+
+          ctx.addIssue({
+            code: z.ZodIssueCode.custom,
+            message: `Larger Disk size of at least ${formatNumber(diskSizeRequiredForIopsWithGp3)} GB required. Current max is ${formatNumber(maxIopsAllowedForDiskSizeWithGp3)} IOPS.`,
```

Comment:
nit: spelling consistency

```suggestion
            message: `Larger disk size of at least ${formatNumber(diskSizeRequiredForIopsWithGp3)} GB required. Current max is ${formatNumber(maxIopsAllowedForDiskSizeWithGp3)} IOPS.`,
```